apps/demos/Demos/Chat/Overview/Angular/app/app.component.ts

```diff
@@ -0,0 +1,93 @@
+import { NgModule, Component, enableProdMode } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+
+import notify from 'devextreme/ui/notify';
+
+import { DxChatModule } from 'devextreme-angular';
+import { User, Message } from 'devextreme/ui/chat';
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+let modulePrefix = '';
+// @ts-ignore
+if (window && window.config.packageConfigPaths) {
+  modulePrefix = '/app';
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: `.${modulePrefix}/app.component.html`,
+  styleUrls: [`.${modulePrefix}/app.component.css`],
+})
+export class AppComponent {
+  date: Date = new Date();
+
+  currentUser: User = {
+    id: "c94c0e76-fb49-4b9b-8f07-9f93ed93b4f3",
+    name: "John Doe",
+  };
+
+  supportAgent: User = {
+    id: "d16d1a4c-5c67-4e20-b70e-2991c22747c3",
+    name: "Support Agent",
+    avatarUrl: "../../../../images/petersmith.png",
+  };
+
+  messages: Message[] = [
```

Comment:
Let's extract it to app.service.ts file. As a example use this demo https://github.com/DevExpress/DevExtreme/blob/9712e9c7ac7b16cf0d5aa7faba77f5704267bdcb/apps/demos/Demos/ButtonGroup/Overview/Angular/app/app.service.ts#L66

Service simply returns an array of data, which we receive in the constructor. https://github.com/DevExpress/DevExtreme/blob/9712e9c7ac7b16cf0d5aa7faba77f5704267bdcb/apps/demos/Demos/ButtonGroup/Overview/Angular/app/app.component.ts#L33-L34

Reply:
60ab7bbfd4ab1378c123a01d624d163ba2fe0486